test/support/src/helpers.cc

```diff
@@ -1129,8 +1129,14 @@ void write_array(
   tiledb_query_t* query;
   rc = tiledb_query_alloc(ctx, array, TILEDB_WRITE, &query);
   CHECK(rc == TILEDB_OK);
-  if (subarray != nullptr) {
-    rc = tiledb_query_set_subarray(ctx, query, subarray);
+  tiledb_subarray_t* subarray;
+  if (sub != nullptr) {
+    rc = tiledb_subarray_alloc(ctx, array, &subarray);
+    REQUIRE(rc == TILEDB_OK);
+    query->query_->set_subarray(sub);
+    rc = tiledb_subarray_set_subarray(ctx, subarray, sub);
+    CHECK(rc == TILEDB_OK);
+    rc = tiledb_query_set_subarray_t(ctx, query, subarray);
     CHECK(rc == TILEDB_OK);
```

Comment:
Same point on REQUIRE. In this case we haven't set up the test array as expected, and so probably shouldn't proceed with the test at all.